tests/testDipoleFitter.py

```diff
@@ -0,0 +1,407 @@
+from lsst.afw.table import (SourceTable, SourceCatalog)
+from lsst.ip.diffim.dipoleFitTask import (DipoleFitAlgorithm, DipoleFitConfig, DipoleFitTask, DipolePlotUtils)
+from lsst.meas.base import SingleFrameMeasurementConfig
+
+# Export DipoleTestUtils to expose fake image generating funcs
+__all__ = ("DipoleTestUtils")
+
+
+class DipoleFitTestGlobalParams(object):
+    """
+    Class to initialize and store global parameters used by all tests below.
+
+    Attributes
+    ----------
+    display : boolean
+        Display (plot) the output dipole thumbnails (matplotlib)
+    verbose : boolean
+        be verbose during fitting
+    w : integer
+        width (pixels) of test generated exposures
+    h : integer
+        height (pixels) of test generated exposures
+    xc : array of floats
+        x coordinate (pixels) of center(s) of input dipole(s)
+    yc : array of floats
+        y coordinate (pixels) of center(s) of input dipole(s)
+    flux : array of floats
+        flux(es) of input dipole(s)
+    gradientParams : array of floats
+        three parameters for linear background gradient
+    offsets : array of floats
+        pixel coordinates between lobes of dipoles
+    """
+
+    def __init__(self):
+        """
+        Initialize the parameters.
+        """
+        np_random.seed(666)
+        self.display = False
+        self.verbose = False
+        self.w, self.h = 100, 100  # size of image
+
+        self.xc = np_array([65.3, 24.2])  # xcenters of two dipoles in image
+        self.yc = np_array([38.6, 78.5])  # ycenters of two dipoles
+        self.flux = np_array([2500., 2345.])  # fluxes of pos/neg lobes
+        self.gradientParams = np_array([10., 3., 5.])
+
+        self.offsets = np_array([-2., 2.])  # pixel coord offsets between lobes of dipoles
+
+
+# First, test the algorithm itself (fitDipole()):
+# Create a simulated diffim (with dipoles) and a linear background gradient in the pre-sub images
+#   then compare the input fluxes/centroids with the fitted results.
+class DipoleFitAlgorithmTest(lsst_tests.TestCase):
+    """
+    A test case for dipole fit algorithm.
+
+    Test the dipole fitting algorithm on two dipoles within an image with
+    simulated noise.
+    """
+    def setUp(self):
+        self.params = DipoleFitTestGlobalParams()
+
+        offsets = self.params.offsets
+        self.dipole, (self.posImage, self.posCatalog), (self.negImage, self.negCatalog) = \
+            DipoleTestUtils.makeDipoleImage(
+                xcenPos=self.params.xc + offsets,
+                ycenPos=self.params.yc + offsets,
+                xcenNeg=self.params.xc - offsets,
+                ycenNeg=self.params.yc - offsets,
+                flux=self.params.flux, fluxNeg=self.params.flux,
+                gradientParams=self.params.gradientParams)
+
+        self.catalog = DipoleTestUtils.detectDipoleSources(self.dipole)
+
+    def tearDown(self):
+        del self.dipole, self.posImage, self.negImage
```

Comment:
One `del` per line please. It makes it clearer in diffs later if items are removed or added.